### PR TITLE
add vyper layout to setBalance

### PR DESCRIPTION
**Pull request by iamchrissmith**
allow setting token balance for vyper storage layout

### CODE REVIEW

**Comment by gbalabasquer** [comment]
The PR looks good, but this won't work for `spells-mainnet` as there, we have a special branch for GUSD.
I want to proceed with this change because is the correct thing, so I guess we have two alternatives:
1) Ignoring completely GUSD in the tests until we need to test something with it and do a specific case.
2) Branching outside the function in all the places where the old giveTokens is being used.

**Review by hexonaut — COMMENTED** [review]
Code looks good. Can you update the comments to say it works for Vyper contracts?

**Comment by hexonaut** [comment]
> The PR looks good, but this won't work for `spells-mainnet` as there, we have a special branch for GUSD. I want to proceed with this change because is the correct thing, so I guess we have two alternatives:
> 
> 1. Ignoring completely GUSD in the tests until we need to test something with it and do a specific case.
> 2. Branching outside the function in all the places where the old giveTokens is being used.

So there are all kinds of edge cases with tokens. I'm okay with adding Vyper support, but I don't think this function should cover all token edge-cases. Beyond GUSD there is also rebase tokens such as stETH, etc.

I think it's ok to have Maker collateral like GUSD in the way it is in spells-mainnet.

**Comment by gbalabasquer** [comment]
> > The PR looks good, but this won't work for `spells-mainnet` as there, we have a special branch for GUSD. I want to proceed with this change because is the correct thing, so I guess we have two alternatives:
> > 
> > 1. Ignoring completely GUSD in the tests until we need to test something with it and do a specific case.
> > 2. Branching outside the function in all the places where the old giveTokens is being used.
> 
> So there are all kinds of edge cases with tokens. I'm okay with adding Vyper support, but I don't think this function should cover all token edge-cases. Beyond GUSD there is also rebase tokens such as stETH, etc.
> 
> I think it's ok to have Maker collateral like GUSD in the way it is in spells-mainnet.

Well but that's the thing, the special branch for GUSD actually happens inside giveTokens, which will be replaced by this function. So the only way we have to cover the case in spells-mainnet is actually branching in each caller of the former giveTokens (which I guess it is fine).

**Comment by iamchrissmith** [comment]
> Code looks good. Can you update the comments to say it works for Vyper contracts?

done

**Review by gbalabasquer — APPROVED** [review]
LGTM

File: src/GodMode.sol [hunk 1]
```diff
@@ -82,33 +82,59 @@ library GodMode {
     }
 
     /// @dev Sets the balance for `who` to `amount` for `token`.
-    /// Note this only works for contracts compiled under Solidity. Vyper contracts use a different storage structure for maps.
-    /// See https://twitter.com/msolomon44/status/1420137730009300992?t=WO2052xM3AzUCL7o7Pfkow&s=19
+    /// Note this only works for contracts compiled under Solidity and Vyper.
```
Comment (hexonaut) [inline src/GodMode.sol:85]: I would just remove this. "only solidity and vyper" I'm not sure what else there is 😅 

Also, can you match the comment for overloaded functions too.

Comment (iamchrissmith) [reply src/GodMode.sol:85]: haha good point, was just "updating" probably don't have to worry too much about yul or ewasm

**Review by hexonaut — APPROVED** [review]
Ok this is good to merge now. 👍
